src/main/java/UseCase/recommendSong.java

```diff
@@ -1,4 +1,55 @@
 package UseCase;
 
-public class recommendSong {
+import Database.songAccessInterface;
+import Database.songDsData;
+import Database.songLibrary;
+import Entities.Song;
+import InputBoundary.recommendationInputBoundary;
+import InputData.playlistInputData;
+import InputData.songInputData;
+import OutputBoundary.songOutputBoundary;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+/**
+ * Application Business Rules layer Use Case Interactor for generating a recommendation based on
+ * a given Playlist.
+ */
+public class recommendSong implements recommendationInputBoundary {
+    songAccessInterface library;
+    songOutputBoundary out;
+    playlistInputData data;
+
+    public recommendSong(playlistInputData data, songOutputBoundary out){
+        this.library = songLibrary.getInstance();
+        this.out = out;
+        this.data = data;
+    }
+
+    /**
+     * Generates a recommendation based on the given Playlist in the constructor.
+     */
+    @Override
+    public void recommendation() {
+        Random random = new Random();
+        ArrayList<Song> playlist = data.getSongs();
+
+        Song randomSong = playlist.get(random.nextInt(playlist.size()));
+        String genre = randomSong.getGenre();
+        Song rec = null;
+
+        for (songDsData song : library.getLibrary()){
+            if (rec == null){
+                rec = song.getSong();
+            }
+            if (genre.equals(song.getGenre()) & !playlist.contains(song.getSong())) {
+                rec = song.getSong();
```

Comment:
Couldn't you just do --

```
rec = song.getSong();
if(genre.equals(rec.getGenre()) && ...) break;
```

?

Reply:
There's also the issue of iterating through the entire songLibrary and not finding a song with the same genre. Were you just gonna do the very last song it grabs from songLibrary? It's fine if you are, just wondering, lol

Reply:
As of right now, if no song is found then it will choose the very first song (the only iteration where rec is null). To address the first point - yeah I could do that _if_ I skipped doing the null check. I figured it would be kind of silly to have two variables iterating through the list (both rec and song, if I remove the null check), which is why I chose for rec to hold the first song in the library until a proper recommendation is found.

Reply:
Ah, I see now what you were trying to do. To be honest, it feels like they have the number of iterations and my suggestion might be clearer to understand for anyone else who'd have to take a look in the future(at least from my initial outside perspective).

I'm fine with not changing it since its functionally the same if you don't want to though.